spec/spec.md

```diff
@@ -0,0 +1,146 @@
+# Lance Catalog Specification
+
+**Lance Catalog** is an open specification on top of the storage-based Lance open table and data format 
+to standardize access to a collection of Lance tables.
+It describes how a catalog service like Apache Hive MetaStore (HMS), Apache Gravitino, Unity Catalog, etc.
+should store and use Lance tables, as well as how ML/AI tools and analytics compute engines
+(will together be called _"tools"_ in this document) should integrate with Lance.
+
+## Definition
+
+A Lance catalog is a centralized repository for discovering, organizing, and managing Lance tables.
+There are 2 types of Lance catalogs: storage directory and REST catalog.
+
+## Lance Directory
```

Comment:
Do we need to use `Lance Directory Catalog` to make it clearer? Let users know it's a `catalog`, not a concept of lance format. WDYT?